Fix issue with parallel derivatives

dschwoerer opened the pull request.

dschwoerer committed:
Fix input for recent parsing update

dschwoerer committed:
Only iterate over valid regions

dschwoerer committed:
add regions to parallel slices

dschwoerer committed:
Merge remote-tracking branch 'panhu/fci-bd' into fci-bd

dschwoerer committed:
only set BOUT_TOP if it is not set

allows to export BOUT_TOP for all projects

dschwoerer committed:
Update printing to fmt

dschwoerer committed:
Update gitignore

dschwoerer committed:
Communicate only once

In case of fci, we need to apply parallel boundaries as well.

dschwoerer committed:
Directly apply parallel boundary

The for loop took a copy, rather then the field itself.

dschwoerer committed:
apply boundaries

dschwoerer committed:
Fix case for options

dschwoerer committed:
Mark some sections conditionally used

dschwoerer committed:
switch to bout-config

allows to  compile with cmake

dschwoerer committed:
update json

dschwoerer committed:
Revert "update json"

This reverts commit fc3fe5d0976cbf7edfa29b7eca3690672d798600.

dschwoerer committed:
add cmake file

dschwoerer committed:
Making simulation more stable

dschwoerer committed:
Do not rely on DC values

DC is inherently unsuitable, as it depends on the free parameter of the
grid generation, namely all parallel slices can be abitrarily rotated in
poloidal direction, thus at most 1D averaging could be used.

dschwoerer committed:
cmake: Set c++ standard

dschwoerer committed:
Avoid deprecation warning

dschwoerer committed:
initialise parallel fields properly

dschwoerer committed:
Don't take DC component but full field

Works, no reason to average

dschwoerer committed:
Some changes

* Always use floor'ed fields
* some additional calls to communicate + parallel BCs

dschwoerer committed:
Don't try to delete a shared_ptr

dschwoerer committed:
fix diffusion2d for FCI

dschwoerer committed:
Add new input file

dschwoerer committed:
Fix parallel slices

dschwoerer committed:
Remove debugging vars

Can be done with boutcore in post processing

dschwoerer committed:
minor cleanup

dschwoerer committed:
More communcations removed

This prevents overwriting the boundary conditions of Te etc for the
kappa_{i,e}par, thus probably more correct then the previous version.

dschwoerer committed:
Start switching to outer loops

dschwoerer committed:
Fix typo in input file

dschwoerer committed:
Switch to BOUT_FOR in diffusion2d

dschwoerer committed:
Improvements from my STORM version

dschwoerer committed:
less communications

dschwoerer committed:
Try to write better omp code

dschwoerer committed:
minor optimisation

dschwoerer committed:
Add faster mul_all

However, changes the result. Not sure why ...

dschwoerer committed:
Move to outer loops

dschwoerer committed:
Fix DO_ALL

dschwoerer committed:
more outer loop

dschwoerer committed:
one more outer loop

dschwoerer committed:
simplify parallel boundary conditions

Unify forward + backward, only difference is the sign in a few cases.

dschwoerer committed:
Only apply sheath BC to outer boundaries

bendudson commented:
Are sheath boundary conditions not needed on all parallel boundaries? Is this to prevent issues in the inner (core) boundary?

dschwoerer commented:
For the W7X mesh the outer boundaries end on the wall, while the inner are towards the hot plasma of the core. Of course they could be overwritten, but I think it makes sense to directly only apply these BCs to the field lines ending up on targets.

dschwoerer committed:
Change solver

dschwoerer committed:
Add CI workflow

dschwoerer committed:
use custom checkout

dschwoerer committed:
Update to bout enum class

dschwoerer committed:
Update input files

dschwoerer committed:
Add grids via subrepository

dschwoerer committed:
CI: run only short

dschwoerer committed:
fix grid to tracked grid

dschwoerer committed:
Ensure field is allocated for copying

dschwoerer committed:
Merge pull request #2 from dschwoerer/ci

Add basic CI workflow

dschwoerer committed:
Input cleanup

* Fix typo (parallel_sheath**s**)
* Suggestions from v5-auto-updater

dschwoerer committed:
more conditionally used

dschwoerer committed:
Merge branch 'fci-ds' of github.com:dschwoerer/hermes-2 into changes

dschwoerer committed:
more conditionally used

dschwoerer committed:
apply fci boundaries

simple low order

dschwoerer committed:
Move flux boundary conditions to separate function

dschwoerer committed:
Remove ion viscosity code

Probably not correct. We can implement something correct later - it
should not be of major important.

dschwoerer committed:
Apply more parallel boundaries

If we apply standard boundaries, we probably also want FCI ones

dschwoerer committed:
Remove non fci code: sheath BC

dschwoerer committed:
Remove non-fci code

dschwoerer committed:
Merge pull request #3 from dschwoerer/changes

* Use correct terms for laplacian newsolver
* Cleanup input files

dschwoerer committed:
Merge pull request #4 from dschwoerer/cleanup

Cleanup + boundary conditions
